cancelreader_test.go

```diff
@@ -0,0 +1,84 @@
+package cancelreader
+
+import (
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestReader(t *testing.T) {
```

Comment:
This test is probably expected to fail on Windows currently, where we only support `os.Stdin`.